hathor/consensus/block_consensus.py

```diff
@@ -47,6 +47,19 @@ def log(cls) -> Any:
 
     def update_consensus(self, block: Block) -> None:
         self.update_voided_info(block)
+        self.assert_valid_consensus(block)
+
+    def assert_valid_consensus(self, block: Block) -> None:
+        """Assert that all transactions confirmed by the block are valid.
+
+        This assertion might be slow because it runs in O(n) where n is number of transactions confirmed
+        by the block."""
```

Comment:
Maybe we should start running some nodes with [`-O`](https://docs.python.org/3/using/cmdline.html#cmdoption-O). It removes `assert` statements, so we could have more liberty in adding costly assertions. We should keep some nodes without `-O` so we do catch bugs when those asserts are reached, but some other nodes could run without them so they don't have a performance penalty.

*The loop, or the whole function, would have to be moved to `if __debug__`.